src/assume.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_ASSUME_H
+#define BITCOIN_ASSUME_H
+
+#include <tinyformat.h>
+
+// ASSUME(expression)
+// ==================
+// ASSUME(...) is used to document assumptions.
+//
+// Bitcoin Core is always compiled with assertions enabled. assert(...)
+// is thus only appropriate for documenting critical assumptions
+// where a failed assumption should lead to immediate abortion also
+// in production environments.
```

Comment:
It's not really clear to me when you should choose `assert`  vs `ASSUME` -- neither is any quicker because both always run the test, and if the assumptions you had when writing the code were wrong, how confident can you really be that continuing won't cause horrible corruption to decide that continuing is okay?